Kernel/Arch/aarch64/PageDirectory.cpp

```diff
@@ -5,35 +5,120 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/Singleton.h>
+#include <Kernel/Arch/CPU.h>
+#include <Kernel/Arch/PageDirectory.h>
 #include <Kernel/Arch/aarch64/ASM_wrapper.h>
-#include <Kernel/Memory/PageDirectory.h>
+#include <Kernel/InterruptDisabler.h>
+#include <Kernel/Memory/MemoryManager.h>
+#include <Kernel/Prekernel/Prekernel.h>
+#include <Kernel/Process.h>
+#include <Kernel/Random.h>
+#include <Kernel/Sections.h>
+#include <Kernel/Thread.h>
 
 namespace Kernel::Memory {
 
-void PageDirectory::register_page_directory(PageDirectory*)
+struct TTBR0Map {
```

Comment:
it’d be way more readable if you decided on something more abstract and descriptive in the naming; i’d propose to limit `ttbr0` usage to cpu register accessors… (i.e. only to `Aarch64::Asm::×××`)

Reply:
I kinda agree, however this code is all very much architecture specific anyway, so in my opinion it is fine to use the architecture specific name, similar to the x86_64 code that uses the CR3 naming.